apps/ios/Shared/Views/Chat/ChatView.swift

```diff
@@ -454,50 +454,55 @@ struct ChatView: View {
         @Published var date: Date?
         @Published var isDateVisible: Bool = false
         var isReallyNearBottom: Bool { scrollOffset.value > 0 && scrollOffset.value < 500 }
-        let visibleItems = PassthroughSubject<[String], Never>()
+        let listState = PassthroughSubject<ListState, Never>()
         let scrollOffset = CurrentValueSubject<Double, Never>(0)
+
         private var bag = Set<AnyCancellable>()
 
+        typealias ListState = (
+            topItemDate: Date?,
+            bottomItemId: ChatItem.ID?
+        )
+
+        private struct ViewUpdate: Equatable {
```

Comment:
`ViewUpdate` struct added to satisfy`.removeDuplicates()` requirement for `Equatable` conformance.